test/VoucherHub.test.ts

```diff
@@ -968,6 +968,119 @@ describe('VoucherHub', function () {
         });
     });
 
+    describe('Drain voucher', function () {
+        let [voucherOwner1, anyone]: SignerWithAddress[] = [];
+        let voucherHub: VoucherHub;
+        let voucher: Voucher;
+        let voucherAddress: string;
+
+        beforeEach(async function () {
+            ({ voucherHub, voucherOwner1, anyone } = await loadFixture(deployFixture));
+            // Create voucher type
+            await voucherHubWithAssetEligibilityManagerSigner
+                .createVoucherType(description, duration)
+                .then((tx) => tx.wait());
+            // Create voucher
+            voucherAddress = await voucherHubWithVoucherManagerSigner
+                .createVoucher(voucherOwner1, voucherType, voucherValue)
+                .then((tx) => tx.wait())
+                .then(() => voucherHub.getVoucher(voucherOwner1));
+            voucher = Voucher__factory.connect(voucherAddress, anyone);
+        });
+
+        it('Should drain all funds of expired voucher', async function () {
+            const voucherHubRlcBalanceBefore = await iexecPocoInstance.balanceOf(voucherHubAddress);
+            // Expire voucher
+            const expirationDate = await voucher.getExpiration();
+            await time.setNextBlockTimestamp(expirationDate + 100n); // after expiration
+            // Drain
+            await expect(voucherHubWithAnyoneSigner.drainVoucher(voucherAddress))
+                .to.emit(iexecPocoInstance, 'Transfer')
+                .withArgs(voucherAddress, voucherHubAddress, voucherValue)
+                .to.emit(voucherHub, 'Transfer')
+                .withArgs(voucherAddress, ethers.ZeroAddress, voucherValue)
+                .to.emit(voucherHub, 'VoucherDrained')
+                .withArgs(voucherAddress, voucherValue);
+            expect(await iexecPocoInstance.balanceOf(voucherAddress))
+                .to.equal(await voucherHub.balanceOf(voucherAddress))
+                .to.equal(await voucher.getBalance())
```

Comment:
```suggestion
```
I think we have another test where we check "hub.balance(voucher) equals voucher.balance()"

Reply:
Can you point it to me please?

Reply:
[Yes here](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/blob/develop/test/beacon/Voucher.test.ts#L323-L325) !

Reply:
Here we are comparing `voucher.getBalance()` before and after.

Reply:
I don't follow you.. let it be :)

Reply:
Which hub are you talking about in `hub.balance(voucher)`?
I'm talking about **VoucherHub**.

Reply:
Wrong thread here. Answered in the correct one.